$OpenBSD: patch-timing_cc,v 1.3 2014/09/25 17:33:15 sthen Exp $

Avoid float-related assertion triggered on i386, not triggered on
macppc/amd64/sparc64.

--- timing.cc.orig	Sat Aug 16 03:45:47 2014
+++ timing.cc	Wed Sep 24 13:01:27 2014
@@ -126,6 +126,8 @@
 
 /* $Id: patch-timing_cc,v 1.3 2014/09/25 17:33:15 sthen Exp $ */
 
+#include <limits>
+
 #include "timing.h"
 #include "NmapOps.h"
 #include "utils.h"
@@ -436,7 +438,7 @@ void RateMeter::update(double amount, const struct tim
     interval = MAX(current_rate_history, diff);
   else
     interval = TIMEVAL_SUBTRACT(*now, start_tv) / 1000000.0;
-  assert(diff <= interval);
+  assert(diff <= interval + std::numeric_limits<float>::epsilon());
   /* If we record an amount in the very same instant that the timer is started,
      there's no way to calculate meaningful rates. Ignore it. */
   if (interval == 0.0)
